resolwe/storage/management/commands/compare_models_and_csv.py

```diff
@@ -0,0 +1,350 @@
+                    next_in_models, model_hash = URLS.next()  # advance models
+                elif next_in_models > next_in_csv or not URLS.has_next():
+                    # entries are missing in models
+                    # (models are alphabetically *ahead*)
+                    fullpath = subpath + "/" + next_in_csv
+                    logger.warning(f"CSV-ONLY {fullpath}")
+                    counter["csv_only"] += 1
+                    next_in_csv, csv_hash = CSV.next()  # advance CSV
+
+            # either (or both) of the iterators is finished,
+            # now we need to exhaust the other
+            while next_in_csv:
+                logger.warning(f"CSV-ONLY {subpath}/{next_in_csv}")
+                counter["csv_only"] += 1
+                next_in_csv, csv_hash = CSV.next()
+            while next_in_models:
+                logger.warning(f"MODEL-ONLY {subpath}/{next_in_models}")
+                counter["models_only"] += 1
+                next_in_models, model_hash = URLS.next()
+
+        # list all subpaths from CSV that we haven't visited
+        # while traversing models' data
+        for subpath in subpath_map:
+            if "visited" not in subpath_map[subpath]:
+                filecount = subpath_map[subpath]["linecount"]
+                logger.warning(f"CSV-ONLY {subpath}/* ({filecount} files)")
+                counter["csv_only"] += filecount
+
+        # print an overview/summary
+        out = ""
+        out += f"{counter['match']} files OK"
+        if counter["csv_only"] != 0:
+            out += f", {counter['csv_only']} files in CSV only"
+        if counter["models_only"] != 0:
+            out += f", {counter['models_only']} files in models only"
+        if counter["hash_mismatch"] != 0:
+            out += f", {counter['hash_mismatch']} files do not match the hash"
+        logger.info(out)
+
+        # double check the numbers just in case
+        ReferencedPath_count = ReferencedPath.objects.exclude(
+            Q(path__endswith="/")
+        ).count()
+        logger.debug(f"CSV length = {CSV.length}")
+        logger.debug(f"ReferencedPath object count = {ReferencedPath_count}")
+
+        matches = counter["hash_mismatch"] + counter["match"]
+        csv_records = matches + counter["csv_only"]
+        models_records = matches + counter["models_only"]
+        # this should never happen, but it's better to check,
+        # just because it's so easy to do
+        if csv_records != CSV.length:
+            logger.debug(
+                "Numbers don't add up."
+                " OK + csv_only + hash_mismatch != CSV.line_count."
+            )
+        if models_records != ReferencedPath_count:
+            logger.debug(
+                "Numbers don't add up."
+                " OK + models_only + hash_mismatch != ReferencedPath_count."
+                " There are probably orphaned ReferencedPaths."
+            )
+
+
+def parseline(line):
```

Comment:
I suggest using ``namedtuple`` (https://docs.python.org/3/library/collections.html#collections.namedtuple) for storing parsed line data. 

Then referencing the parts of the line becomes easier and methods ``get_filename``,  ``get_subpath`` and ``get_etag`` are not needed anymore.